internal/credsretriever/refreshing_cache.go

```diff
@@ -52,6 +54,18 @@ type internalClock func() time.Time
 // type assertion
 var _ credentials.CredentialRetriever = &cachedCredentialRetriever{}
 
+var (
+	promCacheNonRecoverableError = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "pod_identity_cache_non_recoverable_error",
+		Help: "Removing credentials from cache, got non recoverable error",
+	})
+	promCacheHit = promauto.NewCounterVec(prometheus.CounterOpts{
```

Comment:
nit: would suggest to rename to `promCacheState` as one of the state is hit?

Reply:
good catch, fixed :)